### DISCUSSION TITLE
Accesso da wifi a home assistant su Docker

### DISCUSSION THREAD

**Discussion by zuccospooky** [indomus/forum#1628] [Networking]
ciao a tutti, vi seguo da poco e sono alle primissime armi con la domotica. Sto facendo qualche tentativo giusto per provare un po' le funzionalità di home assistant e seguendo la vostra guida ho installato sul mio PC tradizionale home assistant core via docker. Scusate la domanda probabilmente super banale. Sul mio PC ho sua una scheda wifi che una scheda ethernet. Io sono connesso alla rete tramite wifi, il mio PC ha un insirizzo IP locale sulla mia rete wifi e fin qui tutto ok. MA il mio PC ha anche un indirizzo IP diverso legato alla mia scheda ethernet, anche se non ho nessuna periferica ethernet connessa. Il mio ip legato alal scheda wifi è del tipo 192.168 ecc, quello della scheda ethernet è 172.30 ecc. Ora dopo l'installazione tramite docker, io posso accedere al HA su browser digitando il mio IP locale della ethernet e la porta 8123. Questo credo che faccia si che il mio HA non veda nulla di quello che c'è connesso alla mia rete wifi. 
Avete per caso idea di come fare a far comunicare HA con la mia rete wifi? 
Grazie mille!
Fabio

**Comment by reiner348** [comment]
Ciao, purtroppo non posso aiutarti ma visto che la domanda l'hai posta ormai oltre un mese fa, mi sento di darti un consiglio... Ho fatto il tuo stesso errore all'inizio: provare home assistant su PC "per vedere com'è" con l'idea di passare al raspberry. E' un errore perchè ti scontri con problemi, tipo il tuo, incomprensibili a chi è agli inizi (e anche dopo) quasi sempre generati dal fatto stesso di usare windows.
Investi in un raspberry e smanetta su quello! Vedrai che tutto si semplifica e anche quando ti imbatterai in problemi, avrai le risposte quando chiedi qui nella community!

**Reply by snakuzzo** [reply]
Sbagliato IMHO...non serve investire prima di farsi un'idea.
Se proprio si vuole, si può creare una VM linux e fare tutto da lì...non cambia assolutamente niente.

Quanto alla richiesta di @zuccospooky l'ip di classe 172 è quello relativo alla network virtuale di docker. E' normale che sia così.
Il comportamento che hai è normalissimo...ed è dovuto alla isolation di docker.
Su Windows (come su macos) purtroppo hai per forza di cose questo comportamento perché non puoi impostare il [network mode](https://docs.docker.com/network/#network-drivers) host. Il network mode host lo puoi impostare solo su Linux. 

![1](https://user-images.githubusercontent.com/7205371/170983195-3b8fe047-5daf-4c73-8ffe-bb94770f3c48.png)

Questo su HA non è per forza un problema, non ti funzionerà il discovery, ma potrai comunque aggiungere integrazioni e configurare l'aggiunta di device a mano.

P.S. che si sia alle prime armi oppure no, il concetto è sempre lo stesso...tocca leggere e studiare. Volere la roba che funzioni con uno schiocco di dita...è un approccio completamente sbagliato. Se si vuole ciò, basta farsi la domotica con Google Assistant, Siri o Alexa.
